code/game/objects/items.dm

```diff
@@ -19,6 +19,8 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	/// The icon for holding in hand icon states for the right hand.
 	var/righthand_file = 'icons/mob/inhands/items_righthand.dmi'
 
+	var/supports_variations = null //This is a bitfield that defines what variations exist for bodyparts like Digi legs.
```

Comment:
This is bad. 
You're adding a variable to every single item in the game for the _two subtypes_ that use it. 
Additionally, having bodyparts and clothing use the exact same flag is also pretty clunky when each subtype has it's own snowflake code for handling it with no handling on the very type it's defined on. 
Even if you made an exclusive variable for each type you now have two entirely different systems for clothing variation that accomplish the same thing (jumpskirts still use subtypes, despite being the same type of variation), but with the new one _only_ working for snowflake lizard variations.

Reply:
- You're adding a variable to every single item in the game for the two subtypes that use it.

Clothing is not always a subtype of /clothing/ which requires this to be item-level.

- Having bodyparts and clothing use the exact same flag is also pretty clunky when each subtype has it's own snowflake code for handling it with no handling on the very type it's defined on.

I don't actually know what you're talking about here. 

- Even if you made an exclusive variable for each type you now have two entirely different systems for clothing variation that accomplish the same thing (jumpskirts still use subtypes, despite being the same type of variation), but with the new one only working for snowflake lizard variations.

Jumpskirts aren't a compatibility sprite. They're a piece of unique clothing. And sure, right NOW its only used for lizards, but the entire point is it will be used on other content, most obviously, Grod.

Reply:
> Jumpskirts aren't a compatibility sprite. They're a piece of unique clothing. And sure, right NOW its only used for lizards, but the entire point is it will be used on other content, most obviously, Grod.

jumpskirts are a variation of jumpsuits, the same as digigrade suits. There's no reason why they couldn't use the same (or a slightly modified) version as them

> I don't actually know what you're talking about here.

Basically another byproduct of making it item level. The same flag means something entirely different depending on what it's applied to (bodypart, clothing)

> Clothing is not always a subtype of /clothing/ which requires this to be item-level.

You can just add a variable to the clothing type exceptions too

Reply:
`jumpskirts are a variation of jumpsuits, the same as digigrade suits. There's no reason why they couldn't use the same (or a slightly modified) version as them`

Again, no they aren't. Skirts are an *entirely different typepath*, and have their own "object" in game. The lizard sprites do not exist as their own datum, as that would be a massive waste. 

`Basically another byproduct of making it item level. The same flag means something entirely different depending on what it's applied to (bodypart, clothing)`

Ah okay, I *get* what you're saying here, but I dont really agree with it. Anyone could look at the code and see supports_variation is only used on clothing.

`You can just add a variable to the clothing type exceptions too` 
There are so many snowflakey wearables this isnt really reasonable.

Reply:
...Elon, I legitimately have no idea why you're complaining about adding new variables to the item superclass when there's already a solid 5+ others that are already there.

```
body_parts_covered         Only useful for clothing
flags_cover                Only useful for clothing
gas_transfer_coefficient   Only useful on mask type items
allowed                    Only useful for suit type items
breakouttime               Only useful for cuffs
```

The justification is valid. Jumpskirts are not a *AUTOMATIC VARIATION* of jumpsuits, the digitigrade compatibility sprites *ARE*.

Reply:
> Jumpskirts are not a AUTOMATIC VARIATION of jumpsuits, the digitigrade compatibility sprites ARE.

It wouldn't take a whole lot of effort to move them into one system but I'll concede on this point

> ...Elon, I legitimately have no idea why you're complaining about adding new variables to the item superclass when there's already a solid 5+ others that are already there.

Shitcode doesn't justify shitcode, we should be improving variable bloat _not_ making it worse, we can leech off TG's optimizations all we want but that only works so long as we're not actively digging our tech debt hole even deeper

> There are so many snowflakey wearables this isnt really reasonable.

and almost none of them (if any) have or need race variations